canonical-web-and-design/maas-ui
# [hardware sync] - settings - verify api integration

## Done

- update to read and write hours / minutes / seconds in systemd.time timespan format correctly
  - add `timeSpanToDuration`, `timeSpanToSeconds`, `timeSpanToMinutes` utils
- add e2e test for deploy configuration 

## API spec
### config.list
#### WebSocket Response result - new “hardware_sync_interval” option

```
[{
    name: "hardware_sync_interval",
    value: String in systemd time span format  e.g. “15m”
               (only hours, minutes and seconds are recognised)
               https://www.freedesktop.org/software/systemd/man/systemd.time.html 
},…]
```

  -  hardware_sync_interval is set to `15m` by default
### config.update
#### WebSocket Request params - new “hardware_sync_interval” param
```
params: {
   name: "hardware_sync_interval",
   value: String in systemd time span format  e.g. “15m”
               (only hours, minutes and seconds are recognised)
               https://www.freedesktop.org/software/systemd/man/systemd.time.html 
  }
```

Full spec in [MA026 Updating Hardware Info Post Deployment](https://docs.google.com/document/d/1LaL9KuQlyoO2mXQESGh_iEqBDgpvmP-L9vajrLSCwiQ/edit#heading=h.ndji0f9ytkh0)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to user settings -> deploy
- If you've never changed the hardware sync value, verify that the input field have a value of `15`
- Change the hardware sync interval and save the form
- Go to the a machine that can be deployed, e.g. `MAAS/r/machine/c7mn8b/summary` on bolla
- Go to Actions -> deploy and verify that a correct number of minutes with a `min` suffix is displayed

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/795

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
